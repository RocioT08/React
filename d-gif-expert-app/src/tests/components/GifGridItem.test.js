import React from "react";
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";
import "@testing-library/jest-dom";

describe("GifGridItem.test", () => {
  const imgTitle = "Animated GIF";
  const imgUrl =
    "https://media3.giphy.com/media/PMKXvSyDWwQc8/giphy.gif?cid=b1d1d056uwtfhachwkjcdo0og4d4m8990ued0ypenf37rghl&rid=giphy.gif";
  const wrapper = shallow(<GifGridItem title={imgTitle} url={imgUrl} />);

  test("should display <GifGridItem> correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should contains a paragraph with the title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(imgTitle);
  });

  test("should contain the image equals to url and alt", () => {
    const img = wrapper.find("img");

    //console.log(img.html()); // saca el html
    //console.log(img.props()); // saca las propiedades del componente recuperado como un obj
    //console.log(img.prop("src")); // saca el valor de la prop indicada

    expect(img.prop("src")).toBe(imgUrl);
    expect(img.prop("alt")).toBe(imgTitle);
  });

  test("should contains the animate__fadeIn class", () => {
    const div = wrapper.find("div");
    //console.log(div.props());
    const className = div.prop("className");
    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
