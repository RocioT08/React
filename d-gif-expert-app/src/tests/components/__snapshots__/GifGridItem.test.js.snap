// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem.test should display <GifGridItem> correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Animated GIF"
    src="https://media3.giphy.com/media/PMKXvSyDWwQc8/giphy.gif?cid=b1d1d056uwtfhachwkjcdo0og4d4m8990ued0ypenf37rghl&rid=giphy.gif"
  />
  <p>
    Animated GIF
  </p>
</div>
`;
